.github/workflows/test.yml

```diff
@@ -103,8 +105,21 @@ jobs:
       # Otherwise, we'll use the casettes to make the tests fast.
       - name: Test with pytest using casettes
         if: ${{ github.event_name != 'schedule' }}
+        # We'll also generate code coverage, which we will later turn into a
+        # diff report if this pipeline is running on a pull request.
+        run: |
+          python -m pytest --record-mode=none --cov=allspice --cov-report xml:coverage.xml
+      - name: Generate coverage report
+        if: ${{ github.event_name == 'pull_request' }}
         run: |
-          python -m pytest --record-mode=none
+          git fetch origin ${{ github.base_ref }} --depth=2
+          diff-cover coverage.xml --compare-branch=origin/${{ github.base_ref }} --markdown-report coverage.md
```

Comment:
Do we need to use a software package for this? Or do we actually need diffing on it in general?

Reply:
The package correlates the XML coverage report with the diff to show the coverage gaps in only code changed in the PR. I think that makes it a little easier to identify gaps in what you're working on in the moment, since we don't have perfect coverage on the project as of now. If we don't want a comment on the repo, I can make a separate requirements file and a script to do this purely locally.